services/efs-guardian/tests/unit/test_efs_removal_policy_task.py

```diff
@@ -0,0 +1,175 @@
+
+
+from collections.abc import AsyncIterator
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Any
+from unittest.mock import MagicMock, patch
+
+import pytest
+from faker import Faker
+from fastapi import FastAPI
+from models_library.users import UserID
+from pytest_simcore.helpers.monkeypatch_envs import setenvs_from_dict
+from pytest_simcore.helpers.postgres_tools import insert_and_get_row_lifespan
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from simcore_postgres_database.models.projects import projects
+from simcore_postgres_database.models.users import users
+from simcore_postgres_database.utils_repos import transaction_context
+from simcore_service_efs_guardian.core.settings import (
+    ApplicationSettings,
+    AwsEfsSettings,
+)
+from simcore_service_efs_guardian.services.background_tasks import removal_policy_task
+from simcore_service_efs_guardian.services.efs_manager import (
+    EfsManager,
+    NodeID,
+    ProjectID,
+)
+
+pytest_simcore_core_services_selection = ["postgres", "redis"]
+pytest_simcore_ops_services_selection = [
+    "adminer",
+]
+
+
+@pytest.fixture
+def app_environment(
+    monkeypatch: pytest.MonkeyPatch,
+    app_environment: EnvVarsDict,
+    postgres_env_vars_dict: EnvVarsDict,
+    with_disabled_rabbitmq_and_rpc: None,
+    wait_for_postgres_ready_and_db_migrated: None,
+    with_disabled_redis_and_background_tasks: None,
+):
+    # set environs
+    monkeypatch.delenv("EFS_GUARDIAN_POSTGRES", raising=False)
+
+    return setenvs_from_dict(
+        monkeypatch,
+        {
+            **app_environment,
+            **postgres_env_vars_dict,
+            "POSTGRES_CLIENT_NAME": "efs-guardian-service-pg-client",
+            "EFS_REMOVAL_POLICY_TASK_AGE_LIMIT_TIMEDELTA": "3600",
+        },
+    )
+
+
+@pytest.fixture
+async def user_in_db(
+    app: FastAPI,
+    user: dict[str, Any],
+    user_id: UserID,
+) -> AsyncIterator[dict[str, Any]]:
+    """
+    injects a user in db
+    """
+    assert user_id == user["id"]
+    async with insert_and_get_row_lifespan(  # pylint:disable=contextmanager-generator-missing-cleanup
+        app.state.engine,
+        table=users,
+        values=user,
+        pk_col=users.c.id,
+        pk_value=user["id"],
+    ) as row:
+        yield row
+
+
+@pytest.fixture
+async def project_in_db(
+    app: FastAPI,
+    user_in_db: dict,
+    project_data: dict[str, Any],
+    project_id: ProjectID,
+) -> AsyncIterator[dict[str, Any]]:
+    """
+    injects a project in db
+    """
+    assert f"{project_id}" == project_data["uuid"]
+    async with insert_and_get_row_lifespan(  # pylint:disable=contextmanager-generator-missing-cleanup
+        app.state.engine,
+        table=projects,
+        values=project_data,
+        pk_col=projects.c.uuid,
+        pk_value=project_data["uuid"],
+    ) as row:
+        yield row
+
+
+@patch("simcore_service_efs_guardian.services.background_tasks.get_redis_lock_client")
```

Comment:
Actually this is a very clear and light way to patch functions on a test. Thx!

Reply:
THOUGHT: instead of using `from unittest.mock` I would consider using `pytest_mock` and also make it more explicty since `pytest_mock.patch(...)`